### PR TITLE
Support TypeScript

### LINKED ISSUE #3: Support TypeScript projects by adding type declaration
**Which problem is this feature request solving?**
- Support TypeScript - it doesn't right now. I'm trying out in a TS project and it throws error since it couldn't find the type def file for this package.
   ```ts
   import { getBinPath } from 'get-bin-path';

   const binPath = await getBinPath();
   ```
- This is the error you'll get. 
   image

**Describe the solution you'd like**
- Add type declaration file.

**Can you submit a pull request?**
- Yes

PR done #4

### PR DESCRIPTION
**Pull request by kabirbaidhya**
**Which problem is this pull request solving?**

Resolves #3 

**Checklist**

- [x] I have read the [contribution guidelines](../blob/master/CONTRIBUTING.md).
- [x] Usage is same as JavaScript so no change in the documentation required.
- [x] I have added tests (we are enforcing 100% test coverage).
- [x] The status checks are successful (continuous integration). Those can be
      seen below.

### CODE REVIEW

**Comment by kabirbaidhya** [comment]
@ehmicky

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/**
+ * Options for the getBinPath() function.
+ */
+export interface BinaryOptions {
+  name?: string;
```
Comment (ehmicky) [inline types/index.d.ts:11]: Would it be possible to add descriptions and default values on each option?
You can re-use the ones from the `README`.

File: types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * @example
+ *  const binaryPath = await getBinPath()
+ */
+export function getBinPath(options?: BinaryOptions): Promise<string | undefined>;
+
+/**
+ * Get the current package's binary path synchronously.
+ *
+ * @param {BinaryOptions} [options={}]        Additional options for the binary.
+ * @returns {(Promise<string | undefined>)}
```
Comment (ehmicky) [inline types/index.d.ts]: This does not return a `Promise`.

File: package.json [hunk 3]
```diff
@@ -55,10 +58,12 @@
   "devDependencies": {
     "@ehmicky/dev-tasks": "^0.31.44",
     "ava": "^3.8.2",
+    "dtslint": "^3.4.0",
     "gulp": "^4.0.2",
     "husky": "^4.2.5",
     "pkg-dir": "^4.2.0",
-    "test-each": "^2.0.1"
+    "test-each": "^2.0.1",
+    "typescript": "^3.8.3"
```
Comment (ehmicky) [inline package.json]: Could you please update to the latest version of `typescript` and `dtslint`?

**Review by ehmicky — CHANGES_REQUESTED** [review]
Thanks a lot for this PR @kabirbaidhya!
There are few minor changes, but overall it looks good!

**Comment by ehmicky** [comment]
@all-contributors Could you please add @kabirbaidhya for adding this code?

**Comment by kabirbaidhya** [comment]
@ehmicky Done.

File: types/index.d.ts [hunk 4]
```diff
@@ -2,7 +2,18 @@
  * Options for the getBinPath() function.
  */
 export interface BinaryOptions {
+  /**
+   * Name of the binary. Only needs to be specified when the package exports several binaries.
+   *
+   * Default: `package.json` `name` property
```
Comment (ehmicky) [inline types/index.d.ts]: Would it be possible to use JSDoc? Thanks :)

**Comment by ehmicky** [comment]
Thanks :)
Also please make sure the CI tests are passing (more info in [`CONTRIBUTING.md`](https://github.com/ehmicky/get-bin-path/blob/master/CONTRIBUTING.md#development-process)).

File: types/index.d.ts (hunk 4 above)
Comment (kabirbaidhya) [reply types/index.d.ts]: Updated.

**Comment by kabirbaidhya** [comment]
@ehmicky Fixed. Build mostly all passed - but the last one has an error I don't have any idea on :slightly_smiling_face:. Perhaps a rebuild will fix.

**Review by ehmicky — APPROVED** [review]

**Comment by ehmicky** [comment]
Thanks a lot for your work @kabirbaidhya!

**Comment by ehmicky** [comment]
I have just released it in [`5.1.0`](https://github.com/ehmicky/get-bin-path/releases/tag/5.1.0).

**Pull request merged**